satpy/readers/viirs_sdr.py

```diff
@@ -319,7 +320,7 @@ def concatenate_dataset(self, dataset_group, var_path):
         if variable.size != scans.size:
             for gscans in scans.values:
                 data_chunks.append(self[var_path].isel(y=slice(start_scan, start_scan + gscans * scan_size)))
-                start_scan += scan_size * 48
+                start_scan += gscans * scan_size
```

Comment:
I don't think this is right. The `_scan_size` method is the number of rows per scan (either 32 or 16). This `48` here is the *number of scans* which I *think* is each value in the `scans` list above.

Reply:
I'm pretty sure this should be one chunk per granule (not per scan), so the start of the next chunk should be incremented by the size of the granule, which is the scan size (16 or 32) times the number of scans in this granule (`gscans`)

Reply:
:facepalm: didn't see you swapped the variables. Yes `gscans` should be correct. Sounds good.